### PR TITLE
fix: Clippy warnings

### PR DESCRIPTION
**Pull request by NicholasLYang**
### Description

We have a lot of clippy warnings. Applied `cargo clippy --fix` and manually fixed some other warnings

### CODE REVIEW

File: crates/turbopack-ecmascript-plugins/src/transform/swc_ecma_transform_plugins.rs [hunk 1]
```diff
@@ -29,21 +29,8 @@ pub struct SwcPluginModule(
...
-                CompiledPluginModuleBytes::from(RawPluginModuleBytes::new(
-                    plugin_name.to_string(),
-                    plugin_bytes,
-                ))
-            },
-            #[cfg(not(feature = "swc_ecma_transform_plugin"))]
-            0: (),
-        }
+    pub fn new(_plugin_name: &str, _plugin_bytes: Vec<u8>) -> Self {
+        Self(())
```
Comment (mehulkar) [inline crates/turbopack-ecmascript-plugins/src/transform/swc_ecma_transform_plugins.rs:33]: what happened here? Is this a safe change?

File: crates/turborepo-lib/src/run/pipeline.rs [hunk 2]
```diff
@@ -33,17 +33,17 @@ struct TaskOutputs {
...
+    Hash,
+    // New will show all new task output and turbo-computed task hashes for cached
     // output
-    NewTaskOutput,
-    // ErrorTaskOutput will show task output for failures only; no cache miss/hit messages are
+    New,
+    // Error will show task output for failures only; no cache miss/hit messages are
     // emitted
-    ErrorTaskOutput,
+    Error,
```
Comment (mehulkar) [inline crates/turborepo-lib/src/run/pipeline.rs:46]: curious, what was clippy's issue with these constants? doesn't seem like a formatting issue

**Review by mehulkar — APPROVED** [review]
Looks like some changes to turbopack code too, might be a good idea to get a look from someone more familiar! 

Also, are we going to put clippy autofix into a precomit hook or fail CI with it or something?

File: crates/turbopack-ecmascript-plugins/src/transform/swc_ecma_transform_plugins.rs (hunk 1 above)
Comment (ForsakenHarmony) [reply crates/turbopack-ecmascript-plugins/src/transform/swc_ecma_transform_plugins.rs:33]: I don't think so

File: crates/turborepo-lib/src/run/pipeline.rs (hunk 2 above)
Comment (ForsakenHarmony) [reply crates/turborepo-lib/src/run/pipeline.rs:46]: Probably the common postfix

**Review by ForsakenHarmony — CHANGES_REQUESTED** [review]

File: crates/turborepo-lib/src/run/pipeline.rs (hunk 2 above)
Comment (NicholasLYang) [reply crates/turborepo-lib/src/run/pipeline.rs:46]: yeah common prefix

File: crates/turbopack-ecmascript-plugins/src/transform/swc_ecma_transform_plugins.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ForsakenHarmony) [reply crates/turbopack-ecmascript-plugins/src/transform/swc_ecma_transform_plugins.rs:45]: ah wait it was missing the underscore prefix, sorry

File: crates/turborepo-lib/src/daemon/server.rs [hunk 3]
```diff
@@ -47,14 +47,17 @@ use crate::{
 };
 
 pub struct DaemonServer<T: Watcher> {
+    #[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (gsoltis) [reply crates/turborepo-lib/src/daemon/server.rs:50]: We currently have the Go daemon enabled, this code gets used again with a feature flag change

**Review by gsoltis — APPROVED** [review]

**Review by sokra — APPROVED** [review]

**Review by ForsakenHarmony — APPROVED** [review]

**Pull request merged**